### PR TITLE
Flatten cell neighbours structure

### PR DESCRIPTION
**Pull request by mpuccio**
@mconcas, this does not change anything (nor the output nor the time, nor memory consumption) but it is required for the next PR that will be more consistent.

### CODE REVIEW

**Review by mconcas — APPROVED** [review]

File: Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx [hunk 1]
```diff
@@ -394,8 +395,13 @@ void TrackerTraits::findCellsNeighbours(const int iteration)
...
     int layerCellsNum{static_cast<int>(mTimeFrame->getCells()[iLayer].size())};
     const int nextLayerCellsNum{static_cast<int>(mTimeFrame->getCells()[iLayer + 1].size())};
-    mTimeFrame->getCellsNeighbours()[iLayer].resize(nextLayerCellsNum);
 
+    mTimeFrame->getCellsNeighboursLUT()[iLayer].clear();
+    mTimeFrame->getCellsNeighboursLUT()[iLayer].resize(nextLayerCellsNum, 0);
+    std::vector<std::pair<int, int>> cellsNeighbours;
+    cellsNeighbours.reserve(nextLayerCellsNum);
+
+    std::vector<std::vector<int>> easyWay(nextLayerCellsNum);
```
Comment (ktf) [inline Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx:404]: @mpuccio in my profiles I see this line as responsible of 80% of the malloc / free calls for its-tracker. Don't we have any other way of doing the same, without the churn?

Comment (mpuccio) [reply Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx:404]: Sure, this is actually a debug vector that I forgot to remove. I will remove it later in the morning

Comment (ktf) [reply Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx:404]: @mpuccio Apparently it's still there. Is there any issue with removing it?

Comment (mpuccio) [reply Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx:404]: I removed the `easyWay` vector; do you want to remove cellsNeighbours? This is needed; I can add a reserve if you see that this slows down the processing (I honestly did not see that).

Comment (ktf) [reply Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx:404]: No, i guess I simply mis remembered which vector. Sorry for the noise. I will profile again an let you know if anything else is needed.

**Pull request merged**